### PR TITLE
Add possibility to correctly inject service manager to the ZF form, and process the input filters and elements.

### PR DESCRIPTION
**Pull request by spalax**

### CODE REVIEW

**Comment by Xapp73** [comment]
[Block spam users](https://trello.com/c/naPRHKNu/909-block-spam-users)

File: src/Form/UserForm.php [hunk 1]
```diff
@@ -209,108 +213,104 @@ protected function addElements()
...
+                    ['name' => 'StringTrim'],
                 ],
                 'validators' => [
                     [
                         'name' => 'StringLength',
                         'options' => [
-                            'min' => 6,
-                            'max' => 64
+                            'min' => 1,
+                            'max' => 128
```
Comment (ikovalyov) [inline src/Form/UserForm.php:231]: email column in db is varchar(255)

File: src/Form/UserForm.php [hunk 2]
```diff
@@ -209,108 +213,104 @@ protected function addElements()
...
+                    'name' => 'password',
+                    'required' => true,
+                    'filters' => [
+                    ],
+                    'validators' => [
+                        [
+                            'name' => 'StringLength',
+                            'options' => [
+                                'min' => 6,
+                                'max' => 64
```
Comment (ikovalyov) [inline src/Form/UserForm.php:281]: password is varchar (255) in db

**Comment by ikovalyov** [comment]
👍 just a few comments

File: src/Form/UserForm.php (hunk 2 above)
Comment (spalax) [reply src/Form/UserForm.php:281]: legacy

File: src/Form/UserForm.php (hunk 1 above)
Comment (spalax) [reply src/Form/UserForm.php:231]: legacy

File: src/Form/UserForm.php (hunk 2 above)
Comment (ikovalyov) [reply src/Form/UserForm.php:281]: @spalax I can't get it. Why did you add 6-64 characters constraint here?

Comment (spalax) [reply src/Form/UserForm.php:281]: @ikovalyov i did not it is you did, or somebody else, it is legacy, i've added merge_recursive.

Comment (ikovalyov) [reply src/Form/UserForm.php:281]: @spalax ok, I created a follow-up ticket so that w ewon't forget to fix it later

**Pull request merged**